Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -1186,14 +1156,18 @@ extension SwiftLanguageServer {
     return codeActions
   }
 
-  func retrieveQuickFixCodeActions(_ params: CodeActionRequest) -> [CodeAction] {
-    guard let cachedDiags = currentDiagnostics[params.textDocument.uri] else {
+  func retrieveQuickFixCodeActions(_ params: CodeActionRequest) async throws -> [CodeAction] {
+    let diagnosticsResponse = try await self.documentDiagnostic(
+      DocumentDiagnosticsRequest(
+        textDocument: params.textDocument
+      )
+    )
+    guard case .full(let diagnosticsReport) = diagnosticsResponse else {
+      log("Expected a full diagnostic report")
       return []
     }
 
-    let codeActions = cachedDiags.flatMap { (cachedDiag) -> [CodeAction] in
-      let diag = cachedDiag.diagnostic
-
+    let codeActions = diagnosticsReport.items.flatMap { (diag) -> [CodeAction] in
```

Comment:
I assume the idea here is that we have an AST in sourcekitd anyway, so no need to cache? How often are the quick fix suggestions requested?

Reply:
Yes, that was my thought, sourcekitd caches the AST and there’s not that much communication going on since we only return the diagnostics (which should be very few for a single file, typically).

They are requested whenever you move the cursor in VSCode (like cursor info in Xcode). I just checked the logs and getting the diagnostics takes ~2ms. I don’t think it’s worth adding caching of the diagnostics for that.

Reply:
That's a fair bit of log spam/extra processing for every cursor move though, especially when it's not really necessary. 2ms is quite high for that IMO.

Reply:
I’ll open a follow-up PR to introduce caching and then we can actually measure how much difference it makes to decide if we want it.